v2/pkg/runner/enumerate_test.go

```diff
@@ -0,0 +1,45 @@
+package runner
+
+import (
+	"github.com/stretchr/testify/require"
+	"os"
+	"testing"
+)
+
+func TestFilterAndMatchSubdomainForMatch(t *testing.T) {
+	options := &Options{}
+	options.Domain = []string{"example.com"}
+	options.Threads = 10
+	options.Timeout = 10
+	options.Output = os.Stdout
+	options.Match = []string{"*.example.com"}
```

Comment:
I think we should cover a few more cases in detail:
- Literals (e.g. `reg.example.com`)
- Multiple wildcards (e.g. `*.api-*`)
- Sequential Cascade Matchers
- Sequential Cascade Filters
- Matchers together with filters

Reply:
Done